templates/server/src/main/resources/archetype-resources/api/src/test/java/__packageInPathFormat__/ToTest.java

```diff
@@ -0,0 +1,295 @@
+package ${package};
+
+import java.io.Serializable;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.Year;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.assertj.core.api.SoftAssertions;
+import org.junit.Test;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AssignableTypeFilter;
+import org.springframework.util.ReflectionUtils;
+
+import com.devonfw.module.basic.common.api.reflect.Devon4jPackage;
+import com.devonfw.module.basic.common.api.to.AbstractTo;
+import com.devonfw.module.test.common.base.ModuleTest;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.EqualsVerifierApi;
+import nl.jqno.equalsverifier.Warning;
+
+public class ToTest extends ModuleTest {
+
+  private static final ToExclusions EXCLUSIONS = new ToExclusions();
+
+  static {
+    // EXCLUSIONS.add(MyBusinessEto.class).ignoreEquals().ignoreProperties("foo", "bar");
+  }
+
+  /**
+   * Finds all {@link AbstractTo Transportobjekte} of this app via reflection and tests all getters/setters as well as equals and hashCode.
+   */
+  @Test
+  public void testTos() {
+
+    List<Class<?>> toClasses = getToClasses();
+    SoftAssertions assertion = new SoftAssertions();
+    for (Class<?> toClass : toClasses) {
+      ToExclusion exclusion = EXCLUSIONS.get(toClass.getName());
+      testGetSet(toClass, assertion, exclusion);
+      testEqualsAndHashcode(toClass, assertion, exclusion);
+    }
+    assertion.assertAll();
+  }
+
+  /**
+   * Checks for an overridden equals() method of a given class object
+   */
+  private boolean checkForCustomEquals(Class<?> clazz) {
+
+	for(Method method : clazz.getMethods()) {
+	  if(method.toString().equals("public boolean java.lang.Object.equals(java.lang.Object)")) return false;
```

Comment:
simply use `clazz.getDeclaredMethod("equals", new Class[] { Object.class})` instead.
BTW: Your formatter is entirely broken. We are trying to enforce standard formatting rules so that we do not end up in git diff wars when people do changes and reformat code. Please fix your IDE to be compliant with devonfw.

Reply:
Thanks for the comments. I'll add the dependency to the test module. Should I still fix the methods that check for custom equals() / hashCode() implementations? As far as I can see, AbstractTo only overrides the toString() method, so a class not overriding equals() and hashCode() is possible.